MSAL/test/unit/utils/MSIDTestURLSession+MSAL.m

```diff
@@ -25,62 +25,21 @@
 //
 //------------------------------------------------------------------------------
 
-#import "NSDictionary+MSALTestUtil.h"
+#import "MSIDTestURLSession+MSAL.h"
+#import "MSALTestURLSessionDataTask.h"
 
-@implementation NSDictionary (MSALTestUtil)
+@implementation MSIDTestURLSession (MSAL)
 
-- (BOOL)compareToActual:(NSDictionary *)dictionary
-{
-    BOOL fSame = YES;
-    
-    for (NSString *key in self)
-    {
-        id myVal = self[key];
-        id otherVal = dictionary[key];
-        if (!otherVal)
-        {
-            NSLog(@"\"%@\" missing from result dictionary.", key);
-            fSame = NO;
-        }
-        else if (![myVal isKindOfClass:[MSALTestSentinel class]] && ![myVal isEqual:otherVal])
-        {
-            NSLog(@"\"%@\" does not match. Expected: \"%@\" Actual: \"%@\"", key, self[key], otherVal);
-            fSame = NO;
-        }
-    }
-    
-    for (NSString *key in dictionary)
-    {
-        if (!self[key])
-        {
-            NSLog(@"Extra key \"%@\" in result dictionary: \"%@\"", key, dictionary[key]);
-            fSame = NO;
-        }
-    }
-    
-    return fSame;
-}
-
-- (NSString *)base64UrlJson
-{
-    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:self options:0 error:nil];
-    return [NSString msidBase64UrlEncodeData:jsonData];
-}
+#pragma mark - DataTask creation
 
-@end
-
-@implementation MSALTestSentinel
-
-static MSALTestSentinel *s_sentinel = nil;
-
-+ (void)initialize
+- (NSURLSessionDataTask *)dataTaskWithRequest:(NSURLRequest *)request
```

Comment:
I'm kinda surprised this isn't available in the bigger MSIDTestURLSession, maybe just move it there?

Reply:
In MSIDTestURLSession we have similar method, but instead of block we use delegate. I suggest to replace/add it to common MSIDTestURLSession in separate PR.